cpp/models/seir_ide/seir_ide.h

```diff
@@ -0,0 +1,128 @@
+    *   we need values from time -(m_k-1)*dt until 0 for our calculation (with m_k=std::ceil((m_infectious_time+m_latency_time)/dt)) 
+    *   -> first initial time point should be earlier than -(m_k-1)*dt
+    * @param dt_init size of the time step used for numerical integration.
+    * @param N_init Population of the considered region. 
+    */
+    IdeModel(TimeSeries<double>&& init, double dt_init, int N_init);
+
+    /**
+    * Change m_latency_time, default value: 3.3
+    * @param latency The value of the new latency time
+    */
+    void set_latency_time(double latency);
+
+    /**
+    * Change m_infectious_time, default value: 8.2
+    * @param infectious The value of the new infectious time
+    */
+    void set_infectious_time(double infectious);
+
+    /**
+    * Get Reference to used ContactMatrix (effective Contactfrequency is used = quantity of Contacts * probability of infection in case of contact)
+    */
+    ContactMatrix& get_contact_matrix();
+
+    /**
+    * Simulation of the course of infection
+    * @param t_max last simulation day
+    * @return result of the simulation, stored in a TimeSeries with simulation time and associated number of susceptibles (S).
+    */
+    const TimeSeries<double>& simulate(int t_max);
+
+    /**
+    * Calculate the distribution of the population in E,I and R based on the calculated values for S.
+    * Here average values are calculated using the average latency and infectious time and not the distributions 
+    * actually used by the model. (because they are not explicitly calculable)
+    * @return result of the calculation stored in an TimeSeries. The TimeSeries contains the simulation time and an associated Vector 
+    * with values for S, E, I and R 
+    */
+    const TimeSeries<double>& calculate_EIR();
+
+    /**
+    * print simulation result
+    * @param calculated_SEIR if the values for E,I and R are calculated before with function "calculate_EIR",
+    * one may want to print out these values too.
+    */
+    void print_result(bool calculated_SEIR = false) const;
+
+private:
+    /**
+    * Density of the generalized beta distribution used for the function f_{beta} of the IDGL model.
+    * 
+    * @param tau evaluation point of the generalized Beta distribution
+    * @param p parameter p of the generalized Beta distribution
+    * @param q parameter q of the generalized Beta distribution
+    */
+    double generalized_beta_distribution(double tau, double p = 3.0, double q = 10.0) const;
+
+    /**
+    * Numerical differentiation of one compartment using a central difference quotient.
+    * @param idx Time index at which the numerical differentiation should be performed.
+    * @param compartment Index of the compartment in m_result for which the numerical differentiation is to be performed.
+    * @return compartment'(t[idx]) numerically approximated
+    */
+    double central_difference_quotient(Eigen::Index compartment, Eigen::Index idx) const;
```

Comment:
Thank you for the first change. I already like it much better. My suggestion is to further extend this to:

`central_difference_quotient(TimeSeries<double> const& ts_ide, Eigen::Index compartment, Eigen::Index idx) const;`

And then you can hand over the result itself.